app/domain/authentication/o_auth/discover_identity_provider.rb

```diff
@@ -34,7 +34,7 @@ def discover_provider
         @discovered_provider = @open_id_discovery_service.discover!(@provider_uri).tap do
           @logger.debug(Log::IdentityProviderDiscoverySuccess.new)
         end
-      rescue HTTPClient::ConnectTimeoutError => e
+      rescue HTTPClient::ConnectTimeoutError, Errno::ETIMEDOUT => e
```

Comment:
Do we have a test for these two?

Reply:
We don't have tests for 504 as we had problems with this in the past. I will create an issue for this.

Reply:
As discussed in person we need to investigate this issue further and design a test to mock a 504 GatewayTimeout and not a 502 BadGateway response